torch/csrc/api/include/torch/nn/module.h

```diff
@@ -476,6 +476,31 @@ class TORCH_API Module : public std::enable_shared_from_this<Module> {
       std::string name,
       ModuleHolder<ModuleType> module_holder);
 
+  /// Replaces a registered submodule with this `Module`.
+  ///
+  /// This takes care of the registration, you should assign the submodule
+  /// as well. It only works when a module of the name is already registered.
```

Comment:
It would be great to explicitly say "...you should assign the submodule as well. For example, if you have `l1` as a class member that points to this submodule, you should do both registration and assignment in one line: `model.l1 = model.replace_module("l1", torch::nn::Linear(3,  4))`"